### ISSUE TITLE
CloudHSM 5 pkcs11 library fails when opening a session

### ISSUE THREAD

**Issue by urcheon** [OpenSC/libp11#412]
When using Openssl 1.1.1k, libp11 0.4.11 and CloudHSM pkcs11 client 5.1.0, trying to load a certificate which fails.

I see the following errors in /opt/cloudhsm/run/cloudhsm-pkcs11.log.2021-06-23:

```
2021-06-23T11:24:38.628Z DEBUG [2517] ThreadId(3) [cloudhsm_pkcs11::session::C_OpenSession] C_OpenSession invoked
2021-06-23T11:24:38.628Z TRACE [2517] ThreadId(3) [cloudhsm_pkcs11::session::C_OpenSession] Successfully retrieved session manager
2021-06-23T11:24:38.628Z DEBUG [2517] ThreadId(3) [cloudhsm_pkcs11::slots::slot_manager] Retrieved slot with id 2305843009213693953
2021-06-23T11:24:38.628Z TRACE [2517] ThreadId(3) [cloudhsm_pkcs11::session::C_OpenSession] Retrieved slot with ID 2305843009213693953
2021-06-23T11:24:38.628Z TRACE [2517] ThreadId(3) [cloudhsm_pkcs11::session::C_OpenSession] Session creation option is ReadOnly
2021-06-23T11:24:38.628Z ERROR [2517] ThreadId(3) [cloudhsm_pkcs11::session::C_OpenSession] IE000: Failed to open provider session: ReadOnly cannot be created by Hsm1 provider
2021-06-23T11:24:38.628Z ERROR [2517] ThreadId(3) [cloudhsm_pkcs11::session::C_OpenSession] C_OpenSession failed, returning 0x00000006

```
It seems like CloudHSM client 5 doesn't like to be opened with ReadOnly. 
Same code works fine with CloudHSM 3.3.1.

**Comment by urcheon** [comment]
* ReadOnly issue fixed with CloudHSM version 5.5.0

* FORCE_LOGIN needs to be the last engine parameter

* If you are running a single HSM device in the cluster, you will need to disable key availability check with:
/opt/cloudhsm/bin/configure-pkcs11 --disable-key-availability-check

**Comment by mtrojnar** [comment]
Thank you for the details.  They will be useful for other CloudHSM users.
